src/main/java/org/threadly/litesockets/UDPClient.java

```diff
@@ -180,6 +176,31 @@ public boolean setSocketOption(final SocketOption so, final int value) {
   public ClientOptions clientOptions() {
     return uso;
   }
+  
+  @Override
+  protected void addReadBuffer(final ByteBuffer bb) {
+    addReadStats(bb.remaining());
+    synchronized(readerLock) {
+      readBuffers.add(bb);
+      callReader();
```

Comment:
it wont be, call reader does a .call() on a ListenerHelper that was set with the KeyedExecuter for that client as the executer.  let me know if thats not right, but I am pretty sure that call is async.